assets/evaluation_on_cloud/environments/evaluations-built-in/context/evaluate_on_data.py

```diff
@@ -77,12 +102,150 @@ def get_evaluator_config(command_line_args):
     return evaluator_config
 
 
-def run_evaluation(command_line_args, evaluators, evaluator_config):
+def create_model_target_and_data_mapping(command_line_args):
+    """Get model target configuration from user input."""
+    if not command_line_args.eval_target:
+        logger.info("No eval_target provided. Returning None.")
+        return (None, None)
+    try:
+        logger.info("Eval_target provided.")
+        target_config = json.loads(command_line_args.eval_target)
+    except json.JSONDecodeError as e:
+        logger.error(f"Invalid JSON in eval_target: {e}")
+        return (None, None)
+
+    model_config = target_config.get(MODEL_CONFIG, {})
+
+    api_key_env = model_config.get(API_KEY, "")
+    api_key_value = os.environ.get(api_key_env.upper(), "")
+    if not api_key_value:
+        logger.warning(f"API key environment variable '{api_key_env.upper()}' is missing or empty!")
+
+    model_config_type = str(model_config.get(TYPE, ""))
+    logger.info(f"  - Type: {model_config_type}")
+
+    endpoint = ""
+    if model_config_type == "1":  # AOAI
```

Comment:
as discussed, if UI can send the url then this validation should go inside the service,